src/cargo/util/toml/embedded.rs

```diff
@@ -229,16 +217,28 @@ fn split_source(input: &str) -> CargoResult<Source<'_>> {
         source.content = content;
     }
 
+    // Experiment: let us try which char works better
+    let tick_char = source
+        .content
+        .chars()
+        .filter(|c| ['`', '#', '-'].contains(c))
+        .next()
+        .unwrap_or('`');
+
     let tick_end = source
         .content
         .char_indices()
-        .find_map(|(i, c)| (c != '`').then_some(i))
+        .find_map(|(i, c)| (c != tick_char).then_some(i))
         .unwrap_or(source.content.len());
     let (fence_pattern, rest) = match tick_end {
         0 => {
             return Ok(source);
         }
         1 | 2 => {
+            if tick_char == '#' {
+                // Attribute
+                return Ok(source);
+            }
             anyhow::bail!("found {tick_end} backticks in rust frontmatter, expected at least 3")
```

Comment:
I think this comment needs updated
```suggestion
            anyhow::bail!("found {tick_end} `{tick_char}` in rust frontmatter, expected at least 3")
```